File: src/main/java/com/njit/dmsd/service/QueryService.java
package com.njit.dmsd.service;

import java.util.List;

import com.njit.dmsd.dto.QueryDto;

/**
 * Created by ashish on 13/5/17.
 */
public interface QueryService {
    QueryDto getQueryById(Integer queryId);
    void saveQuery(QueryDto queryDto);
    List<QueryDto> getAllQueries();
    
    List<List<Object>> getQueryResult(String queryKey);
}

File: src/main/resources/static/js/app/QueryService.js
'use strict'

angular.module('demo.services', []).factory('QueryService',
		[ "$http", "CONSTANTS", function($http, CONSTANTS) {
			var service = {};
			service.getQueryById = function(queryId) {
				var url = CONSTANTS.getQueryByIdUrl + queryId;
				return $http.get(url);
			}
			service.getAllQueries = function() {
				return $http.get(CONSTANTS.getAllQueries);
			}
			service.saveQuery = function(queryDto) {
				return $http.post(CONSTANTS.saveQuery, queryDto);
			}
			service.getQueryResult = function(queryDto) {
				var url = CONSTANTS.getQueryResult + queryDto.queryKey;
				return $http.get(url);
			}
			return service;
		} ]);